v2/devices_test.go

```diff
@@ -0,0 +1,330 @@
+					ClientVersion:             "",
+					Created:                   tailscale.Time{},
+					Expires: tailscale.Time{
+						time.Date(1, 1, 1, 00, 00, 00, 0, time.UTC),
+					},
+					Hostname:          "hello",
+					ID:                "50052",
+					IsExternal:        true,
+					KeyExpiryDisabled: true,
+					LastSeen: tailscale.Time{
+						time.Date(2022, 4, 15, 13, 24, 40, 0, time.UTC),
+					},
+					MachineKey:      "",
+					Name:            "hello.tailscale.com",
+					NodeKey:         "nodekey:30dc3c061ac8b33fdc6d88a4a67b053b01b56930d78cae0cf7a164411d424c0d",
+					OS:              "linux",
+					UpdateAvailable: false,
+					User:            "services@tailscale.com",
+				},
+				{
+					Addresses:                 []string{"100.121.200.21", "fd7a:115c:a1e0:ab12:4843:cd96:6265:e618"},
+					Authorized:                true,
+					BlocksIncomingConnections: false,
+					ClientVersion:             "1.22.2-t60b671955-gecc5d9846",
+					Created: tailscale.Time{
+						time.Date(2022, 3, 5, 17, 10, 27, 0, time.UTC),
+					},
+					Expires: tailscale.Time{
+						time.Date(2022, 9, 1, 17, 10, 27, 0, time.UTC),
+					},
+					Hostname:          "foo",
+					ID:                "50053",
+					IsExternal:        false,
+					KeyExpiryDisabled: true,
+					LastSeen: tailscale.Time{
+						time.Date(2022, 4, 15, 13, 25, 21, 0, time.UTC),
+					},
+					MachineKey:      "mkey:30dc3c061ac8b33fdc6d88a4a67b053b01b56930d78cae0cf7a164411d424c0d",
+					Name:            "foo.example.com",
+					NodeKey:         "nodekey:30dc3c061ac8b33fdc6d88a4a67b053b01b56930d78cae0cf7a164411d424c0d",
+					OS:              "linux",
+					UpdateAvailable: false,
+					User:            "foo@example.com",
+				},
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.Name, func(t *testing.T) {
+			actual := make(map[string][]tailscale.Device)
+
+			assert.NoError(t, tc.UnmarshalFunc(tc.DevicesContent, &actual))
+			assert.EqualValues(t, tc.Expected, actual["devices"])
+		})
+	}
+}
+
+func TestClient_DeleteDevice(t *testing.T) {
+	t.Parallel()
+
+	client, server := NewTestHarness(t)
+	server.ResponseCode = http.StatusOK
+	ctx := context.Background()
+
+	deviceID := "deviceTestId"
+	assert.NoError(t, client.Devices().Delete(ctx, deviceID))
+	assert.Equal(t, http.MethodDelete, server.Method)
+	assert.Equal(t, "/api/v2/device/deviceTestId", server.Path)
+}
+
+func TestClient_DeviceSubnetRoutes(t *testing.T) {
+	t.Parallel()
+
+	client, server := NewTestHarness(t)
+	server.ResponseCode = http.StatusOK
+	server.ResponseBody = &tailscale.DeviceRoutes{
+		Advertised: []string{"127.0.0.1"},
+		Enabled:    []string{"127.0.0.1"},
+	}
+
+	const deviceID = "test"
+
+	routes, err := client.Devices().SubnetRoutes(context.Background(), deviceID)
+	assert.NoError(t, err)
+	assert.Equal(t, http.MethodGet, server.Method)
+	assert.Equal(t, "/api/v2/device/test/routes", server.Path)
+	assert.Equal(t, server.ResponseBody, routes)
+}
+
+func TestClient_AuthorizeDevice(t *testing.T) {
```

Comment:
Think we can likely drop this test since it looks like we've removed `AuthorizeDevice`?

Reply:
Good call.

Reply:
Done